src/adaptors/adaptor_listener.c

```diff
@@ -152,12 +150,9 @@ static void _listener_event_handler(pn_event_t *e, qd_server_t *qd_server, void
             if (li->on_accept)
                 li->on_accept(li, pn_event_listener(e), li->user_context);
             else {
-                // the adaptor_listener is closing but a connection attempt arrived before the
-                // cleanup is complete.  Need to accept it then close it when the connection
-                // completes
-                pn_raw_connection_t *close_me = pn_raw_connection();
-                pn_raw_connection_set_context(close_me, &_conn_event_context);
-                pn_listener_raw_accept(pn_event_listener(e), close_me);
+                // The adaptor_listener is closing but a connection attempt arrived before the
```

Comment:
We don't deny connection in the handler of the PN_LISTENER_ACCEPT event in server.c
Should we ?

Reply:
Hrmmm... that server listener has a different lifecycle than the adaptor listener: it stays active until the listener config is explicitly deleted.  That's unlike an adaptor listner which needs to disable/enable the socket frequently depending on the availability of a consumer...

So my feeling is no - this won't happen with server listeners.